### PR TITLE
[Async Refactoring] Get semantics providing expr to decide if call is to completion handler

### PR DESCRIPTION
**Pull request by ahoppen**
Resolves rdar://78011350

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: test/refactoring/ConvertAsync/convert_function.swift [hunk 1]
```diff
@@ -287,15 +287,12 @@ func testReturnHandling2(completion: @escaping (String) -> ()) {
...
-// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=RETURN-HANDLING3 %s
+// RUN: %refactor-check-compiles -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=RETURN-HANDLING3 %s
 func testReturnHandling3(_ completion: (String?, Error?) -> Void) {
   return (completion("", nil))
 }
 // RETURN-HANDLING3:      func testReturnHandling3() async throws -> String {
-// RETURN-HANDLING3-NEXT:   return try await withCheckedThrowingContinuation { continuation in  
-// RETURN-HANDLING3-NEXT:     (continuation.resume(returning: ""))
-// RETURN-HANDLING3-NEXT:   }
+// RETURN-HANDLING3-NEXT:   return ""
```
Comment (hamishknight) [inline test/refactoring/ConvertAsync/convert_function.swift:295]: Could you add a `{{^}}` to this check line just to make sure we don't accidentally insert a double return?

**Review by hamishknight — APPROVED** [review]
Thanks!

File: test/refactoring/ConvertAsync/convert_function.swift (hunk 1 above)
Comment (ahoppen) [reply test/refactoring/ConvertAsync/convert_function.swift:295]: Sure.

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Pull request merged**